samcli/commands/delete/delete_context.py

```diff
@@ -108,38 +109,27 @@ def init_clients(self):
         """
         Initialize all the clients being used by sam delete.
         """
-        if not self.region:
-            if not self.no_prompts:
-                session = boto3.Session()
-                region = session.region_name
-                self.region = region if region else "us-east-1"
```

Comment:
If customer is running this command with `no_prompts` then we are setting region to `us-east-1` by default, is it OK to remove it with this change?

Reply:
My question is why should we default to us-east-1? That doesn't sound reasonable to me. 🤔

Reply:
I do agree this is different than other commands, my vote would be keeping the commands in the same way but since we released `sam delete` with this default, I am not sure if we need to make an announcement or need to keep the backwards compatibility.